<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Title</title>
</head>
<body>
<script>

  function* NinjaGenerator (action) {
    try {
      yield 1;
      fail("此处不应该执行")
    } catch (e) {
      console.log('e =>', e)
    }
  }

  const ninjaIterator = NinjaGenerator("skulk")
  const result1 = ninjaIterator.next();

  console.log(result1.value)

  ninjaIterator.throw("外部向生成器抛出错误信息")

  /**
   * 运行长时间任务也不会发生阻塞
   *  - 将生成器和promise相结合, 调用生成器 next 可以继续唤醒生成器并继续执行
   */

</script>
</body>
</html>
